### DISCUSSION TITLE
tree-shaking

### DISCUSSION THREAD

**Discussion by adrianaris** [Leaflet/Leaflet#8576] [Ideas]
Hi,

The way this library is currently bundled is entirely not tree-shakable when in use with webpack: https://github.com/webpack/webpack/issues/9337#issuecomment-507179765 https://github.com/webpack/webpack/issues/9337#issuecomment-507074474
https://stackoverflow.com/questions/55339256/tree-shaking-with-rollup/72365809#72365809

I propose those modifications: https://github.com/adrianaris/Leaflet/commit/351d199bcc1a309ea463486c008f8185557f76f5

**Comment by jonkoops** [comment]
We're actually already actively working on making the library tree-shakable. We'll be doing a lot of refactoring in v2 to hopefully start making this possible.

**Reply by IvanSanchez** [reply]
See also e.g. #6397 - tree-shaking is (unfortunately) not an issue that can be solved with a small change.

**Reply by jonkoops** [reply]
Yeah, it will not be an easy feat. Also because some of our namespaces rely on each-other. For example: https://leafletjs.com/reference.html#tilelayer-wms

**Comment by adrianaris** [comment]
thanks for the reply,
With the modifications proposed above I drag in 100kB less when importing just Map and Layer into a simple CRA: see this source-map-explorer output: 
![image](https://user-images.githubusercontent.com/6321699/196035392-3c7028a8-e3ac-4f08-8049-c6296b473b20.png)

**Reply by adrianaris** [reply]
source-map-explorer output with npm released leaflet: 
![image](https://user-images.githubusercontent.com/6321699/196035522-2b159317-66a5-48cc-b98f-8d1d35129bf5.png)
